transformer_engine/common/include/transformer_engine/fused_attn.h

```diff
@@ -67,7 +77,51 @@ enum NVTE_QKV_Layout {
                           |   num_heads * head_dim
     \endverbatim
  */
-    NVTE_KV_INTERLEAVED = 2
+    NVTE_KV_INTERLEAVED = 2,
+
+    NVTE_SB3HD = 3,
+    NVTE_SBH3D = 4,
+    NVTE_SBHD_SB2HD = 5,
+    NVTE_SBHD_SBH2D = 6,
+    NVTE_SBHD_SBHD_SBHD = 7,
+    NVTE_BS3HD = 8,
+    NVTE_BSH3D = 9,
+    NVTE_BSHD_BS2HD = 10,
+    NVTE_BSHD_BSH2D = 11,
+    NVTE_BSHD_BSHD_BSHD = 12,
+    NVTE_T3HD = 13,
+    NVTE_TH3D = 14,
+    NVTE_THD_T2HD = 15,
+    NVTE_THD_TH2D = 16,
+    NVTE_THD_THD_THD = 17,
+};
+
+/*! \enum NVTE_QKV_Layout_Group
+ *  \brief Grouping of QKV layouts 
+ */
+enum NVTE_QKV_Layout_Group {
+    /*! 3HD QKV layouts, e.g. BS3HD */
+    NVTE_3HD = 0,
+    /*! H3D QKV layouts, e.g. BSH3D */
+    NVTE_H3D = 1,
+    /*! HD_2HD QKV layouts, e.g. BSHD_BS2HD */
+    NVTE_HD_2HD = 2,
+    /*! HD_H2D QKV layouts, e.g. BSHD_BSH2D */
+    NVTE_HD_H2D = 3,
+    /*! HD_HD_HD QKV layouts, e.g. BSHD_BSHD_BSHD */
+    NVTE_HD_HD_HD = 4,
+};
+
+/*! \enum NVTE_QKV_Format
+ *  \brief Dimension formats for QKV tensors
+ */
+enum NVTE_QKV_Format {
```

Comment:
`NVTE_QKV_Format` and `NVTE_QKV_Layout_Group` are designed to group `NVTE_QKV_Layout` in two different ways - see function `nvte_get_qkv_layout_group` [here](https://github.com/cyanguwa/TransformerEngine/blob/0b40cd1658a8c83cbe8d5d155d11a118be1384d9/transformer_engine/common/fused_attn/fused_attn.cpp#L16C27-L16C27) and function `nvte_get_qkv_format` [here](https://github.com/cyanguwa/TransformerEngine/blob/0b40cd1658a8c83cbe8d5d155d11a118be1384d9/transformer_engine/common/fused_attn/fused_attn.cpp#L47). The name 'format' in `NVTE_QKV_Format` is also similar to the argument `qkv_format` on the Python side (see [here](https://github.com/cyanguwa/TransformerEngine/blob/0b40cd1658a8c83cbe8d5d155d11a118be1384d9/transformer_engine/pytorch/attention.py#L1077C9-L1077C34)).

These groupings are mostly just used to distinguish certain sets or groups of qkv layouts so that we don't need to enumerate them out for a particular action. For example, all `sbhd`-format layouts (`sb3hd`, `sbh3d`, `sbhd_sb2hd`, `sbhd_sbh2d`, `sbhd_sbhd_sbhd`) are supported by the arbitrary_seqlen backend ([here](https://github.com/cyanguwa/TransformerEngine/blob/0b40cd1658a8c83cbe8d5d155d11a118be1384d9/transformer_engine/common/fused_attn/fused_attn.cpp#L136)). Or, all `hd_hd_hd`-group layouts (`sbhd_sbhd_sbhd`, `bshd_bshd_bshd`, `thd_thd_thd`) are using the same commands to create output tensors ([here](https://github.com/cyanguwa/TransformerEngine/blob/0b40cd1658a8c83cbe8d5d155d11a118be1384d9/transformer_engine/pytorch/csrc/extensions/attention.cu#L980)). 

Thanks!